src/hooks/safe-apps/useRemoteSafeApps.ts

```diff
@@ -8,17 +8,18 @@ import useAsync from '../useAsync'
 
 // To avoid multiple simultaneous requests (e.g. the Dashboard and the SAFE header widget),
 // cache the request promise for 100ms
-let cache: Promise<SafeAppsResponse> | undefined
-const cachedGetSafeApps = (chainId: string): ReturnType<typeof getSafeApps> => {
-  if (!cache) {
-    cache = getSafeApps(chainId, { client_url: window.location.origin })
+let cache: Record<string, Promise<SafeAppsResponse> | undefined> = {}
```

Comment:
Had to adjust the cache because it would only do one request on the default chains (gor, mainnet) and remove safes that were previously pinned if those didn't exist on either of the default chains